### ISSUE TITLE
Cannot start server

### ISSUE THREAD

**Issue by BlackRose01** [LsHallo/ets2-dedicated-convoy-server#6]
Hi,

when I want to start a server with your imageI receive the following error. I do not understand where the error comes from.

Could you help me, please?

KR,
BlackRose01

```log
[MP] Server init
[S_API] SteamAPI_Init(): Loaded local 'steamclient.so' OK.
dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory
Setting breakpad minidump AppID = 227300
[MP] LogOn to persistent server account.
CAppInfoCacheReadFromDiskThread took 119 milliseconds to initialize
*** ERROR *** : [MP] Error: Steam log on failed - code 15 .
quit
storing cvars
```

## Docker-compose file
```yml
version: '3'

services:
  ets2:
    image: 'lshallo/ets2-convoy-server'
    container_name: 'ETS2'
    restart: on-failure
    tty: true
    network_mode: host
#    ports:
#      - "27015:27015"
#      - "27016:27016"
#      - "27015:27015/udp"
#      - "27016:27016/udp"
    volumes:
      - "/opt/ets2/server_data:/app"
      - "/opt/ets2/game_data:/root/.local/share/Euro Truck Simulator 2"
    environment:
      - "ETS_SERVER_UPDATE_ON_START=true"
      - "ETS_SERVER_NAME=xxx"
      - "ETS_SERVER_PASSWORD=xxx"
      - "ETS_SERVER_LOGON_TOKEN=<TOKEN CREATED WITH STEAMGAMEMANAGER>"
      - "ETS_SERVER_PLAYER_DAMAGE=false"
      - "ETS_SERVER_HIDE_COLLIDING=false"
      - "ETS_SERVER_SERVICE_NO_COLLISION=true"
      - "ETS_SERVER_MODERATORS=xxx"
```

## Full error log
[https://gist.github.com/BlackRose01/0752347dab6c22d037a62843bf0ee7ff](https://gist.github.com/BlackRose01/0752347dab6c22d037a62843bf0ee7ff)

**Comment by LsHallo** [comment]
Thanks for the report.
If you don't mind please put the full error log into a github gist and link it here to make it easier to read this conversation.

I also get the same error as you. Maybe the server structure changed. I will investigate tomorrow.

**Comment by BlackRose01** [comment]
Thank you. I updated the issue and move the log to a gist.

**Comment by LsHallo** [comment]
I have created a forum topic since I can't find `steamservice.so` anywhere to expand the `LD_LIBRARY_PATH` or create a symlink: https://forum.scssoft.com/viewtopic.php?t=319994

**Comment by LsHallo** [comment]
I may have found a partial fix. To try it for yourself please go as follows:
1. Navigate to the directory where your docker-compose is located.
2. Run `git clone -b fix/steamclient.so --depth=1 https://github.com/LsHallo/ets2-dedicated-convoy-server.git steam-fix`
3. Edit your docker-compose.yml:
  - Comment out `image: lshallo/ets2-convoy-server`
  - Add below:
    ```yaml
        build:
          context: steam-fix
    ```
 - Should look like this:
    ```yaml
       [...]
       ets2-server:
         # image: lshallo/ets2-convoy-server
         build:
           context: steam-fix
         container_name: ets2-server
         restart: unless-stopped
       [...]
    ```
4. Run `docker compose build && docker compose up -d` to start the server.

This however has a new problem that steam login is broken with `error 106`.
I have tried the steamclient.so from the game install folder and the steamclient.so from the steamcmd folder. Both have the same issue.  
You can however start the server anonymously and join via the join token that is displayed. (not tested; should work)

I hope to get an answer in the forums how we can fix this. Maybe steamcmd needs to be updated. I don't know.

**Comment by BlackRose01** [comment]
Thanks for that workaround. It works. When I remove the ETS_SERVER_LOGON_TOKEN and change the Dockerfile FROM part to
```
FROM cm2network/steamcmd:latest
````
then I won't receive the 106 error.

**Comment by LsHallo** [comment]
I think I figured it out.
After the reply in the forum I've created a new logon token as suggested.
I then got error 15 instead of error 106.
This indicated that the token is for the wrong game. (I think, not much docs)
I've accidentally created a token for the server app id (1948160) and not the game app id (227300).
After creating the correct token (and using the new fix branch) I can now log in without issue.
